api/api.proto

```diff
@@ -3522,6 +3542,77 @@ message CloudDriveTransferJob {
   string status = 3;
 }
 
+message CollectDiagsJob {
+  // Request is the user request for this diags collection job
+  SdkDiagsCollectRequest request = 1;
+  // Statuses is a list of statuses for diags collection for each node that is part of the request
+  repeated DiagsCollectionStatus statuses = 2;
+}
+
+message DiagsCollectionStatus {
+  // Node is the node that's collecting the diags
+  string node = 1;
+  // State is the current state of diags collection on the node
+  State state = 2;
+  // Message is a user friendly message for current status of diags collection
+  string message = 3;
+
+  // State is an enum for state of diags collection on a given node
+  enum State {
+    // Unspecified means uninitialized or unknown state
+    UNSPECIFIED = 0;
+    // Pending indicates the diags collection is pending and hasn't started
+    PENDING = 1;
+    // Running indicates diags collection is actively running
+    RUNNING = 2;
+    // Done indicates diags collection has finished
+    DONE = 3;
+    // Failed indicates diags collection has failed
+    FAILED = 4;
+  }
+}
+
+
+message SdkDiagsCollectRequest {
+  // Node selects the node(s) for diags collection
+  DiagsNodeSelector node = 1;
+  // Volume selects the volume(s) for diags collection
+  DiagsVolumeSelector volume = 2;
+  // ProfileOnly is an optional flag if true will only collect the stack and heap profile of the driver and will skip
+  // other diag components
+  bool profile_only = 3;
+  // Issuer is an optional  user friendly name for the caller invoking the API
+  string issuer = 4;
+  // TimeoutMins is the timeout in minutes for the job. This is an optional field and if not provided, the
+  // implementation of the SDK will use a sane default
+  uint64 timeout_mins = 5;
+}
+
+// SdkDiagsCollectResponse defines a response for an SDK request to collect diags
+message SdkDiagsCollectResponse {
+  // Job that was created for the SDK request
+  Job job = 1;
+}
+
+// DiagsNodeSelector allows selecting nodes for diags collection
+message DiagsNodeSelector {
+  // NodeLabelSelector is a label selector used to select the nodes for which diags will be collected
```

Comment:
Should these three be in a `oneof` ? Sounds like not all three should be chosen. If `all` is true, then there is no need to set the others, right? Or if `node_ids` are set, then labels do not need to be used?

Reply:
Added more documentation.